### PR TITLE
Pinning pyyaml to 6.0 for now to get past critical security alert

### PR DESCRIPTION
**Pull request by donmccasland**
### Background 
Updating pyyaml to 6.0 to address dependabot alert

### Change Summary
Pinning pyyaml to 6.0 in requirements.in.  Eventually this will need to be reverted to allow google-python-cloud-debugger to pull in the version it prefers.

### CODE REVIEW

**Review by bourgeoisor — APPROVED** [review]
LGTM. Interesting that the source of `pyyaml`, which is `google-python-cloud-debugger`, hasn't been updated since last summer: https://pypi.org/project/google-python-cloud-debugger/#history

File: src/recommendationservice/requirements.in [hunk 1]
```diff
@@ -8,4 +8,5 @@ opencensus-ext-grpc==0.7.1
 python-json-logger==0.1.11
 google-cloud-profiler==1.1.2
 requests==2.27.1
-urllib3==1.26.5
\ No newline at end of file
+urllib3==1.26.5
+pyyaml==6.0
```
Comment (NimJay) [inline src/recommendationservice/requirements.in:12]: **Nitpick:**
We could leave a comment here similar to
```
pyyaml==6.0 # This is not a direct dependency. We will need to keep an eye on google-python-cloud-debugger's use of pyyaml, update google-python-cloud-debugger, and eventually remove this line.
```
or maybe create a GitHub issue.
I just want to make sure the work that's yet to be done is visible to our team. :)

**Review by NimJay — APPROVED** [review]
Left one not-so-important comment.
But looks good to me. Did a quick test of staging URL, http://http://146.148.38.59/ — more specifically the recommendations ("You May Also Like" section). 
Seems to work fine!  
Approved!

File: src/recommendationservice/requirements.in (hunk 1 above)
Comment (NimJay) [reply src/recommendationservice/requirements.in:12]: I've created a new issue to track this: https://github.com/GoogleCloudPlatform/microservices-demo/issues/713
So we can do ahead and merge this pull-request. :)
Thanks again, Don, for this PR and the thanks, Olivier, for reviewing!

**Pull request merged**